### PR TITLE
Ingest Gaia data script

### PR DESCRIPTION
**Pull request by kelle**

### CODE REVIEW

**Comment by kelle** [comment]
This script successfully works on the .db file to modify records for 1266 sources:
```
Names added to database:  1266
Proper motions added to database:  1076
Parallaxes added to database:  1076
Grp: Photometry measurements added to database:  1106
G: Photometry measurements added to database:  1266
```
Once this PR is approved and merged, I'll do a separate PR modifying the 1266 JSON files.

File: scripts/ingests/utils.py
Comment (dr-rodriguez) [inline scripts/ingests/utils.py:58]: I feel like `db_file` should be a parameter that the user can pass to the `load_simpledb` function, even it has a default value of `'SIMPLE.db'`

File: scripts/ingests/utils.py
Comment (dr-rodriguez) [inline scripts/ingests/utils.py:751]: You can do something like: `db.search_object(source, output_table='Parallaxes')` to return the Parallaxes table for the matched source without having save a `db_name` variable that you use in the subsequent query. You still have access to the source name via the _source_ column in Parallaxes.

Comment (dr-rodriguez) [reply scripts/ingests/utils.py:751]: Actually, I guess you may need to keep `db_name` for ingesting brand new parallaxes so maybe this isn't as useful.

File: scripts/ingests/utils.py
Comment (dr-rodriguez) [inline scripts/ingests/utils.py:764]: You should not need to loop over each result to match references/adopted. Something like `ind = plx_data['reference'] == plx_refs[i]` should give you an array of True/False that you can sum up to see if any matches were found. Similarly for the checking of adopted check, which you can then use with `old_adopted = plx_data[ind]` to store that matched row.

File: scripts/ingests/utils.py
Comment (dr-rodriguez) [inline scripts/ingests/utils.py:780]: Isn't this effectively the same as `old_adopted` above? Or do you have this here so you can check that adopted was reset?

File: scripts/ingests/utils.py
Comment (dr-rodriguez) [inline scripts/ingests/utils.py:795]: You shouldn't need to convert `plx[i]` and similar to strings; leaving them as floats should be fine and would avoid issues were you may have values that can't become strings.

File: scripts/ingests/Gaia/ingest_gaia.py
Comment (dr-rodriguez) [inline scripts/ingests/Gaia/ingest_gaia.py:96]: In development/testing, I understand why you would want to store this to a file, but I don't think this is something that needs to be added to the repo. The Gaia designations for DR2 are not going to change if you run this today, and if this is meant to be re-run we would want to re-create it anyways since we may have more sources in the SIMPLE database. In short: because these xml are dynamically drawn from external sources I would argue that we not store the xml results in this repo and instead have the code to regenerate them (which you do here).

File: scripts/ingests/Gaia/ingest_gaia.py
Comment (dr-rodriguez) [inline scripts/ingests/Gaia/ingest_gaia.py:168]: You should add a few comment lines as to what this is doing, my understanding is that it is finding the best match in cases were there were multiple DR3 sources for the same DR2 source.
I assume magnitude_difference and angular_distance are things returned by the ADQL query above? If so, you might be able to modify the query to include something like `ORDER BY angular_distance DESC, magnitude_difference DESC` and that means the first row result would be likely to be the best match. Though how this works with your `group_py` approach, I'm not sure.

File: scripts/ingests/Gaia/ingest_gaia.py
Comment (dr-rodriguez) [inline scripts/ingests/Gaia/ingest_gaia.py:251]: An alternative to this would have been to modify https://github.com/SIMPLE-AstroDB/SIMPLE-db/blob/main/scripts/ingests/add_photometry_filters.py and add a few extra lines of the form:
```
{'telescope': 'Gaia', 'instrument': 'Gaia', 'filter': 'G'},
{'telescope': 'Gaia', 'instrument': 'Gaia', 'filter': 'Grp'},
```
since that scripts automatically adds missing Telescopes, Instruments, PhotometryFilters, if they don't exist, and it queries the SVO to get the wavelength/width information.
The only thing I see that would need verification is that it expects to construct the band name from the instrument+filter so the `fetch_svo` function would need some minor edits.

**Comment by dr-rodriguez** [comment]
Overall, this looks fine. The comments I had were mainly minor suggestions or thoughts of alternative ways to do things. 

One final suggestion is that for your ingest script example, I would put all the functions up front near the top of the page so that the actual actions of ingest are not broken up by function definitions. That makes the workflow much neater.

**Review by dr-rodriguez — APPROVED** [review]
I'm approving this as-is, but encourage you to check the comments above for ways to improve things. I leave it up to you if you want to modify that for this PR or just consider it for future scripts.

File: scripts/ingests/utils.py
Comment (kelle) [reply scripts/ingests/utils.py:780]: Yep, it is a check to see that the adopted flag has changed.

File: scripts/ingests/utils.py
Comment (kelle) [reply scripts/ingests/utils.py:795]: If we don't convert to strings, it adds numbers to the values. e.g, 45.23 becomes something like 45.23000123. But since the datatype is float in the database, it is ingested as a float even when I convert to a string. So I prefer to do the string thing so that we don't end up with the extra decimals.

File: scripts/ingests/Gaia/ingest_gaia.py
Comment (kelle) [reply scripts/ingests/Gaia/ingest_gaia.py:251]: Huh, ok, I will keep this is mind for future ingests.

File: scripts/ingests/Gaia/ingest_gaia.py
Comment (kelle) [reply scripts/ingests/Gaia/ingest_gaia.py]: I added some comments but like the way that I select the best match explicitly.

File: scripts/ingests/Gaia/ingest_gaia.py
Comment (kelle) [reply scripts/ingests/Gaia/ingest_gaia.py]: I removed the XML files from the PR.

File: scripts/ingests/utils.py
Comment (kelle) [reply scripts/ingests/utils.py]: Ah, excellent -- thanks! Implemented.

File: scripts/ingests/utils.py
Comment (kelle) [reply scripts/ingests/utils.py]: agreed.

**Pull request merged**
